src/server/pkg/sql/sql.go

```diff
@@ -0,0 +1,61 @@
+package sql
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+)
+
+type pgDumpReader struct {
+	schemaHeader []byte
+	rd           *bufio.Reader
+}
+
+func NewPGDumpReader(r *bufio.Reader) *pgDumpReader {
+	return &pgDumpReader{
+		schemaHeader: make([]byte, 0),
+		rd:           r,
+	}
+}
+
+func (r *pgDumpReader) ReadRows(count uint64) (rowsDump []byte, err error) {
+	endLine := "\\."
+	// use r.ReadBytes(delim) to read up until insert
+	// store the header, then read each row until there are no more rows left
+	// The first read needs to populate the header
+	if len(r.schemaHeader) == 0 {
+		done := false
+		for !done {
+			b, err := r.rd.ReadBytes('\n')
+			if err != nil {
+				if err == io.EOF {
+					return nil, fmt.Errorf("file does not contain row inserts")
+				}
+				return nil, err
+			}
+			if strings.HasPrefix(string(b), "COPY") {
```

Comment:
This parsing feels a bit fragile, and writing non-fragile parsing yourself sounds quite annoying. Check this out and see if it's worth using: https://github.com/xwb1989/sqlparser

Reply:
Discussed offline, but for posterity. I tried that lib as well as https://github.com/araddon/qlbridge but neither supported the SQL syntax we need. (Both barfed on `COPY` statements ... and qlbridge didn't like the pgdump format for `CREATE TABLE ...` either).

So for now, we're leaving our very dumb parser in place.